### PR TITLE
Add py39 wheels to travis/azure

### LINKED ISSUE #3056: Provide Python3.9 Wheels
#### Problem description

I'm attempting to install this on Windows as it is a dependency for a program I wish to run however, to install from source Microsoft Visual C++ 14.0 is required. 

Would it be possible to provide wheels for this version? 

#### Steps/code/corpus to reproduce

do `pip install gensim` on Python3.9 Windows 10

```
Microsoft Visual C++ 14.0 is required. Get it with "Build Tools for Visual Studio": https://visualstudio.microsoft.com/downloads/   
```

#### Versions

Windows-10-10.0.19041-SP0
Python 3.9.2 (tags/v3.9.2:1a79785, Feb 19 2021, 13:44:55) [MSC v.1928 64 bit (AMD64)]
Bits 64
NumPy 1.20.1
SciPy 1.6.1
gensim N/A (attempting to install 3.8.3)

Thanks

### PR DESCRIPTION
**Pull request by FredHappyface**
Fixes #3056
Follows from #2966

## Summary 
Added py 3.9 build targets to travis/azure

### CODE REVIEW

**Comment by FredHappyface** [comment]
Error building the nmslib wheel looking at the logs https://pypi.org/project/nmslib/

**Comment by mpenkov** [comment]
Can you please conditionally disable Py3.9 wheel builds on Windows?

We'll have to handle them separately when the nsmlib wheels become available.

**Comment by mpenkov** [comment]
@piskvorky WDYT?

Our options for Python 3.9 are:

1. Build wheels for non-Windows builds
2. Build nothing until nmslib wheels are available

@FredHappyface Please let me know if the above isn't a correct interpretation of the choice we need to make.

**Comment by FredHappyface** [comment]
I would find it useful if we went ahead and built Windows wheels anyway as that is used in a pretty small component if I remember correctly?

But then again I would understand why you might be reluctant to do so. Of those two options I'd be most inclined to go with the former

**Comment by gojomo** [comment]
Indeed if only one narrow function doesn't work, due to some other library not working, it could make sense to conditionally disable the test, make that a new open issue, and add a release-note that says: "NMS-related features don't work under Python 3.9 on Windows due to a problem with the [whatever] project, use an lower Python version if you need those functions."

**Comment by piskvorky** [comment]
Yes. We definitely don't want to drop Windows builds on account of some marginal (in the sense of barely used in Gensim) external library.

**Comment by mpenkov** [comment]
@FredHappyface Are you able to conditionally disable nmslib-dependent code such that Python3.9 builds for Windows succeed?

**Comment by FredHappyface** [comment]
Manged to do that but now stymied by pyemd. Not sure if you want me to push what I've got or hold off for a bit?

**Comment by mpenkov** [comment]
Thank you for taking this up. Can we handle PyEMD in the same way? I understand that the rabbit hole is getting deeper...

**Comment by FredHappyface** [comment]
I'll certainly take a look. And yeah it is

I take it then that it's not essential to core functionality?

**Comment by FredHappyface** [comment]
Further down the rabbit hole we go... https://github.com/ztane/python-Levenshtein/issues/63

**Comment by piskvorky** [comment]
Jesus. What kind of apocalypse happened in Windows Py3.9?

**Comment by FredHappyface** [comment]
No idea - a few things have been deprecated but not sure why it's destroyed so many extensions

**Comment by FredHappyface** [comment]
Looks like we are getting the same error as in the upstream repos on my local machine. (I am certain that MS BuildTools is installed). I have no idea what is causing this

<details>
  <summary>Spoiler warning</summary>

  ```cmd
PS C:\Users\Dell\Documents\GitHub\gensim> tox -e py39-win
py39-win create: C:\Users\Dell\Documents\GitHub\gensim\.tox\py39-win
py39-win installdeps: pip>=19.1.1, .[test-win]
py39-win installed: atomicwrites==1.4.0,attrs==20.3.0,colorama==0.4.4,Cython==0.29.22,gensim @ file:///C:/Users/Dell/Documents/GitHub/gensim,iniconfig==1.1.1,mock==4.0.3,Morfessor==2.0.2a4,numpy==1.20.1,packaging==20.9,pluggy==0.13.1,py==1.10.0,pyparsing==2.4.7,pytest==6.2.2,scipy==1.6.1,smart-open==4.2.0,testfixtures==6.17.1,toml==0.10.2
py39-win run-test-pre: PYTHONHASHSEED='1'        
py39-win run-test: commands[0] | python --version
Python 3.9.2
py39-win run-test: commands[1] | pip --version
pip 21.0.1 from C:\Users\Dell\Documents\GitHub\gensim\.tox\py39-win\lib\site-packages\pip (python 3.9)
py39-win run-test: commands[2] | python setup.py build_ext --inplace
C:\Users\Dell\Documents\GitHub\gensim\.tox\py39-win\lib\site-packages\setuptools\dist.py:461: UserWarning: Normalizing '4.0.0beta' to '4.0.0b0'
  warnings.warn(tmpl.format(**locals()))
running build_ext
building 'gensim.models.word2vec_inner' extension
error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
ERROR: InvocationError for command 'C:\Users\Dell\Documents\GitHub\gensim\.tox\py39-win\Scripts\python.EXE' setup.py build_ext --inplace (exited with code 1)_________________________________________________________________________ summary __________________________________________________________________________
ERROR:   py39-win: commands failed
  ```
  
</details>

**Comment by FredHappyface** [comment]
Ok managed to fix the bug I introduced to py3.8 testing 

And managed to skip tests relying on Levenshtein for py3.9

Happy for me to push this (and is there anyway to disable the tests as I'm assuming these pushes are beginning to use up your azure hours)

**Comment by FredHappyface** [comment]
Think I've sorted this out now.

**Review by mpenkov — CHANGES_REQUESTED** [review]
Thank you for your effort and persistence in getting this to work.

I left you some minor comments. Please have a look and get back to me.

File: gensim/similarities/levenshtein.py [hunk 1]
```diff
@@ -43,8 +43,10 @@ def levdist(t1, t2, max_distance=float("inf")):
         The Levenshtein distance between `t1` and `t2`.
 
     """
-    import Levenshtein
-
+    try:
+        import Levenshtein
+    except ImportError:
+        raise ImportError("Levenshtein not installed. Please run `pip install Levenshtein`.")
```
[the remark this answers is not in the document]
Comment (FredHappyface) [reply gensim/similarities/levenshtein.py:49]: Thanks, my bad there! Slight oversight. Will fix

File: gensim/test/test_similarities.py [hunk 2]
```diff
@@ -1546,6 +1546,10 @@ def test_inner_product_corpus_corpus_true_true(self):
 
 class TestLevenshteinDistance(unittest.TestCase):
     def test_max_distance(self):
+        try:
```
[the remark this answers is not in the document]
Comment (FredHappyface) [reply gensim/test/test_similarities.py:1549]: I agree, this reads far better

File: setup.py [hunk 3]
```diff
@@ -270,20 +270,25 @@ def run(self):
 #    'pytest-rerunfailures',  # disabled 2020-08-28 for <https://github.com/pytest-dev/pytest-rerunfailures/issues/128>
     'mock',
     'cython',
-    'nmslib',
-    'pyemd',
     'testfixtures',
     'Morfessor==2.0.2a4',
+]
+
+not_py39_win_testenv = [
```
[the remark this answers is not in the document]
Comment (FredHappyface) [reply setup.py:277]: Yeah that also makes far more sense. I'll make these changes tomorrow

File: gensim/similarities/levenshtein.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mpenkov) [reply gensim/similarities/levenshtein.py:49]: BTW, I don't think we need this try-except here if we catch the import error in gensim.similarities.__init__.

**Comment by piskvorky** [comment]
Of course! You're on a roll :)

Can you merge `develop` into this PR? There were some changes recently, like removing some of the dependencies such as `viz`. To avoid wasted effort & potential git conflicts.

File: gensim/similarities/__init__.py [hunk 4]
```diff
@@ -3,7 +3,18 @@
...
 # bring classes directly into package namespace, to save some typing
-
+import warnings
+try:
+    import Levenshtein
+except ImportError:
+    warnings.warn(
+        "The gensim.similarities.levenshtein submodule is disabled, because the optional "
+        "Levenshtein package <https://pypi.org/project/python-Levenshtein/> is unavailable. "
+        "Install Levenhstein (e.g. `pip install Levenshtein`) to suppress this warning."
```
Comment (mpenkov) [inline gensim/similarities/__init__.py:13]: ;)

```suggestion
        "Install Levenhstein (e.g. `pip install python-Levenshtein`) to suppress this warning."
```

Comment (FredHappyface) [reply gensim/similarities/__init__.py:13]: Blast! Missed that! 😅 Right today's job

**Comment by mpenkov** [comment]
Merging. Good work @FredHappyface ! Thank you for your contribution.

**Pull request merged**